wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet/Update.hs

```diff
@@ -16,6 +19,16 @@ import           Cardano.Wallet.Kernel.DB.Util.AcidState
   UPDATE
 -------------------------------------------------------------------------------}
 
+-- | Errors thrown by 'updateHdAccountName'
+data UpdateHdAccountError =
+    -- | Parent root not found
+    UpdateHdAccountUnknownRoot UnknownHdRoot
+    -- | Account not found
+  | UpdateHdAccountUnknownAccount UnknownHdAccount
+  deriving Eq
```

Comment:
Why do we need a custom error type here? `UnknownHdAccount` already distinguishes between two cases

```haskell
-- | Unknown account
data UnknownHdAccount =
    -- | Unknown root ID
    UnknownHdAccountRoot HdRootId

    -- | Unknown account (implies the root is known)
  | UnknownHdAccount HdAccountId
  deriving Eq
```

Reply:
I think you are right. Right off the bat I cannot justify why I did that. Will give it a spin in cleaning this up.